modules/_labs/button/react/lib/types.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Standard button Button
+ */
+export enum ButtonVariant {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  OutlinePrimary = 'outlinePrimary',
+  OutlineSecondary = 'outlineSecondary',
+  OutlineInverse = 'outlineInverse',
+}
+export enum ButtonSize {
```

Comment:
There's a bit more duplication with the types. Very little duplication when it comes to variants, but unfortunately since subsets of enums aren't really a thing in TS, I think we will need duplicated sizes.

If we were just using the sizes for interfaces, we would be fine.
```tsx
export interface TextButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  // ...
  size: ButtonSize.Medium | ButtonSize.Small;
  // ...
}
```

The problem comes when we need to export the static member variables.

```tsx
TextButton.Size = ???
```
I don't believe there's a way to indicate a subset of the `ButtonSize` enum here.

Reply:
I guess we could do 
```tsx
export interface TextButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  // ...
  size: ButtonSize.Medium | ButtonSize.Small;
  // ...
}

TextButton.Size = {
  Small: ButtonSize.Small,
  Medium: ButtonSize.Medium,
}
```

Reply:
Turns out this doesn't work. It gets mad when trying to equate `TextButton.Size.Small` with `ButtonSize.Small`. Any ideas to keep this static member functionality and avoid a separate size interface for each button type @NicholasBoll?

Reply:
@anicholls @NicholasBoll would it be better in this case to just use a union of strings rather than an enum? I'm not sure the enum is buying us much here so to make it appease TS just sounds silly at this point.

Reply:
@lychyi So usage would turn into `<Button size="small" />`?

Reply:
Personally I've avoided enums in favor of union of string literals. In this case, enums only give us more typing, not more type safety. Autocomplete is actually more helpful with union of string literals.

It can be introduced as a non-breaking change by converting the type to a union of string literals and the enum to an object with an `as const` or casting manually:

```tsx
export interface TextButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  // ...
  size: 'medium' | 'small';
  // ...
}

TextButton.Size = {
  Small: 'small',
  Medium: 'medium',
} as const

// or
TextButton.Size = {
  Small: 'small' as 'small',
  Medium: 'medium' as 'medium',
}
```

I don't remember if we can't use `as const`. Downlevel DTS should support this

Reply:
`as const` came around in 3.4 so I think we should be good with that 👍